features/largeVideo/actions.js

```diff
@@ -28,10 +28,21 @@ export function selectEndpoint() {
             const videoTrack = getTrackByMediaTypeAndParticipant(
                 tracks, MEDIA_TYPE.VIDEO, largeVideo.participantId);
 
-            conference.selectParticipant(
-                videoTrack && videoTrack.videoType === VIDEO_TYPE.CAMERA
-                    ? largeVideo.participantId
-                    : null);
+            try {
+                conference.selectParticipant(
+                    videoTrack && videoTrack.videoType === VIDEO_TYPE.CAMERA
+                        ? largeVideo.participantId
+                        : null);
+            } catch (ex) {
+                // XXX DataChannels are initialized at some later point when
+                // conference has multiple participants, but this code might
+                // be executed before. So here we're swallowing a particular
+                // error.
+                // TODO this should be fixed in lib-jitsi-meet.
+                if (ex.message !== 'Data channels support is disabled!') {
+                    throw ex;
+                }
```

Comment:
1. This piece of source code is a duplication of the one in features/base/conference/middleware.js. To comply with our coding style (https://github.com/jitsi/jitsi-meet-react/blob/master/doc/coding-style.md), please remove the duplication.

2. The duplicated piece of source code requires special attention as indicated by the XXX comment. To ease the maintenance of the source code, I'd like us to have one XXX rather than two. That's another reason for 1.

3. The duplicated piece of source code has future work planned as indicated by the TODO comment. To ease the maintenance of the source code, I'd like us to have one TODO rather than two. That's another reason for 1.